projects/core/directives/dropdown/dropdown-position.directive.ts

```diff
@@ -44,15 +44,17 @@ export class TuiDropdownPositionDirective extends TuiPositionAccessor {
             top: hostRect.top - 2 * offset - viewport.top,
             bottom: viewport.bottom - hostRect.bottom - 2 * offset,
         } as const;
+        const left =
+            hostRect.left + width < viewport.right - offset ? hostRect.left : right;
         const position = {
             top: hostRect.top - offset - height,
             bottom: hostRect.bottom + offset,
-            right,
+            right: viewport.left > right ? viewport.left : right,
```

Comment:
Should it be `viewport.left + offset`? Also `Math.max(viewport.left + offset, right)`